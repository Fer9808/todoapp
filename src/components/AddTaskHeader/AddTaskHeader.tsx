import React from "react";
import {
    TouchableOpacity
} from 'react-native';
// @ts-ignore
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';

import {
    Container,
    IconContainer
} from './styles'
import Text from '../Text/Text';
import * as ACTIONS from '../../types';


const AddTaskHeader = ({
    onPressAction
} : ACTIONS.AddTaskHeaderInterface) => {
    return (
        <Container>
            <IconContainer>
                <TouchableOpacity
                    //@ts-ignore
                    onPress={onPressAction}
                >
                    <MaterialCommunityIcons name="chevron-left" color={'black'} size={25} />
                </TouchableOpacity>
            </IconContainer>
            <Text
                fontSize={'20px'}
                fontWeight={800}
            >
                Add task
            </Text>
        </Container>
    );
}

export default AddTaskHeader;