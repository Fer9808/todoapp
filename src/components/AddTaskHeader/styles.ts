// @ts-ignore
import styled from 'styled-components/native';

export const Container = styled.View`
    background-color: white;
    flex-direction: row;
    align-items: center;
    borderBottomWidth: 2px;
    border-color: #DBDBDB;
    paddingVertical: 20px;
`

export const IconContainer = styled.View`
    background-color: white;
    flex-direction: row;
    align-items: center;
    marginLeft: 20px;
    marginRight: 20px
`