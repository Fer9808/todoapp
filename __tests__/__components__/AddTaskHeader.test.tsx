import * as React from 'react';
import { render } from '@testing-library/react-native';

import AddTaskHeader from '../../src/components/AddTaskHeader/AddTaskHeader';

jest.mock('react-native-vector-icons/MaterialCommunityIcons', () => 'MaterialCommunityIcons'); 

describe('<AddTaskHeader />', () => {
    const  propsMockup = {
        onPressAction: jest.fn()
    }

    const component = render(<AddTaskHeader { ...propsMockup } />)

    it('Render correctly AddTaskHeader Component', () => {
        expect(component).toMatchSnapshot();
    })
})
