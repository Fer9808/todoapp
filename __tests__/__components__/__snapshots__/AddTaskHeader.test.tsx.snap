// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddTaskHeader /> Render correctly AddTaskHeader Component 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderBottomWidth": 2,
        "borderColor": "#DBDBDB",
        "flexDirection": "row",
        "paddingVertical": 20,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flexDirection": "row",
          "marginLeft": 20,
          "marginRight": 20,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <MaterialCommunityIcons
        color="black"
        name="chevron-left"
        size={25}
      />
    </View>
  </View>
  <Text
    allowFontScaling={false}
    color="black"
    fontSize="20px"
    fontWeight={800}
    style={
      Array [
        Object {
          "color": "black",
          "fontSize": 20,
          "fontWeight": "800",
        },
      ]
    }
  >
    Add task
  </Text>
</View>
`;
